Remove msiexec dependency by adding LessMSI for extracting 7-Zip MSI files

JavierMatosD opened the pull request.

Commit:
stop using 7z.msi

Commit:
add endif()

Commit:
add newline EOF

BillyONeal commented on scripts/cmake/vcpkg_find_acquire_program(7Z).cmake:
```diff
@@ -1,9 +1,8 @@
 set(program_name 7z)
 if(CMAKE_HOST_WIN32)
     set(tool_subdirectory "24.08")
-    set(paths_to_search "${DOWNLOADS}/tools/7zip_msi-${tool_subdirectory}-windows/Files/7-Zip") # vcpkg fetch 7zip_msi path
-    list(APPEND paths_to_search "${DOWNLOADS}/tools/7z/${tool_subdirectory}/Files/7-Zip")
```
I think you still need these.

BillyONeal replied:
Oh, I missed the _msi

Commit:
adjust path to search

Commit:
lets try 7z2408.exe with vcpkg x-extract

Commit:
remove duplicate find_acquire_7z

Commit:
swap the search path

Commit:
print paths_to_search

Commit:
lets try this

JavierMatosD committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg into update_find_acquire_7z

JavierMatosD committed:
add vcpkg_find_acquire_program(LESSMSI).cmake and modify ports

JavierMatosD committed:
version db

JavierMatosD committed:
add newline to vcpkg_find_acquire_program(LESSMSI)

JavierMatosD commented:
The goal is to remove `msiexec` from the critical path.

Initially, I tried switching the `7zip` archive from [7z2408.msi](https://github.com/ip7z/7zip/releases/download/24.08/7z2408.msi) to [7z2408-extra.7z](https://github.com/ip7z/7zip/releases/download/24.08/7z2408-extra.7z), only to find that the `.7z` package is a reduced version of `7-Zip`. Hence, changing to the `.7z` package would result in `vcpkg_find_acquire_program(7Z)` providing the reduced version instead of the full version - which is not what I believe most users are expecting. 

Furthermore, some ports require extracting `.exe` files that act as self-extracting archives, which necessitates the full `7-Zip`. However, the full version of `7-Zip` is only available as an `.msi` or `.exe` installer.

To resolve this:

- Added `lessmsi` to `vcpkgTools.xml`
- Updated the extraction logic in `vcpkg_extract_archive.cmake` to use `lessmsi` for `.msi` files instead of relying on `msiexec`
- Updated all ports using `vcpkg-tool-lessmsi` to call `vcpkg_find_acquire_program(LESSMSI)` for extracting `.msi` packages.

JavierMatosD commented:
This would leave the `vcpkg-tool-lessmsi` completely unused in the registry. Delist?

dg0yt commented:
> This would leave the `vcpkg-tool-lessmsi` completely unused in the registry. Delist?

Hm, weren't versioned tool ports prefered over the unversioned scripts?
If not even intentionally interacting (`vcpkg-tool-lessmsi` dependency + `vcpkg_find_acquire_program(LESSMSI)`)?

BillyONeal commented:
> Hm, weren't versioned tool ports prefered over the unversioned scripts?

This was my understanding. Can we just add the `vcpkg_find_acquire_program` entry and not change everyone else using the thing?

We do need a vcpkg_find_acquire_program because 'scripts' like vcpkg_extract_archive can't depend on ports.

dg0yt commented:
> We do need a vcpkg_find_acquire_program because 'scripts' like vcpkg_extract_archive can't depend on ports.

It is clear that a baseline of extraction tools (e.g. the tools needed to extract lessmsi) cannot depend on ports. This is what I understand as the critical path. Bootstrapping.

But I don't see why ports should't be able to declare dependencies on more specific extractions tools as versioned dependencies, even when used in `vcpkg_extract_archive`. The task is to implement a coupling between `vcpkg_extract_archive`/`vcpkg_find_acquire_program` and the calling ports which allows for discovering and *reasonably reporting a missing declared dependency*.

We already use `vcpkg-port-config.cmake` to provide `LESSMSI`. `vcpkg_find_acquire_program(LESSMSI).cmake` could use that. 
~~~cmake
if(NOT DEFINED LESSMSI)
  message(FATAL_ERROR "${PORT} needs the 'lessmsi' program, but lacks a host dependency on port 'vcpkg-tool-lessmi.')
endif()
~~~

Later, this might be generalized.

dg0yt commented:
Additions to `vcpkg_find_acquire_program` should be reflected in `scripts/test_ports/vcpkg-find-acquire-program`.
